### PR TITLE
CAMEL-11531 - set servlet dependency as provided

### PR DESCRIPTION
**Pull request by onders86**

### CODE REVIEW

**Comment by onders86** [comment]
Merged #1836

**Comment by davsclaus** [comment]
Oh this causes the spring-boot starter to fail.

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:compile (default-compile) on project camel-servlet-starter: Compilation failure
[ERROR] /Users/davsclaus/workspace/camel/platforms/spring-boot/components-starter/camel-servlet-starter/src/main/java/org/apache/camel/component/servlet/springboot/ServletMappingAutoConfiguration.java:[43,16] cannot access javax.servlet.http.HttpServlet
[ERROR]   class file for javax.servlet.http.HttpServlet not found

Maybe that starter need some special logic to mark the servlet api as required or something so it can compile

**Comment by davsclaus** [comment]
Just a heads up, that I reverted this on master branch until we find a working solution.

**Comment by onders86** [comment]
i'll reopen the issue and check.

File: platforms/spring-boot/components-starter/camel-servlet-starter/pom.xml [hunk 1]
```diff
@@ -38,7 +38,19 @@
       <groupId>org.apache.camel</groupId>
       <artifactId>camel-servlet</artifactId>
       <version>${project.version}</version>
+      <exclusions>
+        <exclusion>
+		  <groupId>javax.servlet</groupId>
```
Comment (dmvolod) [inline platforms/spring-boot/components-starter/camel-servlet-starter/pom.xml]: @onders86 it's would be nice to fix xml formatting

Comment (onders86) [inline platforms/spring-boot/components-starter/camel-servlet-starter/pom.xml]: mvn -Psourcecheck did not fail. Nice catch. Thank you.

File: platforms/spring-boot/components-starter/camel-servlet-starter/pom.xml [hunk 2]
```diff
@@ -38,7 +38,19 @@
       <groupId>org.apache.camel</groupId>
       <artifactId>camel-servlet</artifactId>
       <version>${project.version}</version>
+      <exclusions>
+        <exclusion>
+          <groupId>javax.servlet</groupId>
+          <artifactId>javax.servlet-api</artifactId>
+        </exclusion>
+	  </exclusions>
```
Comment (onders86) [inline platforms/spring-boot/components-starter/camel-servlet-starter/pom.xml]: here is another formatting issue. i got it :)

**Comment by davsclaus** [comment]
Okay so this works now, then feel free to merge to master

**Pull request merged**